### ISSUE TITLE
JSON DS file: The anyxml "p" is expected to be represented as JSON name/value, but input data contains name/empty array.

### ISSUE THREAD

**Issue by jktjkt** [CESNET/libyang#1960]
I have some code which pushes large quantities of data into the operational DS, and for performance reasons we use `anyxml` instead of a simple `leaf-list`. When we push an empty array *the second time*, we get the following error message:
```
54.050 [268445 sysrepo E] JSON DS file: The anyxml "p" is expected to be represented as JSON name/value, but input data contains name/empty array.
54.050 [268445 sysrepo E] Callback "load" of plugin "JSON DS file" for module "czechlight-roadm-device" failed.
```
The error is "persistent", it looks like it prevents a module uninstall:
```
+ sysrepoctl --uninstall czechlight-roadm-device
[ERR] Change subscription(s) to module "czechlight-roadm-device" no longer valid.
sysrepoctl error: Failed to uninstall modules (Operation failed)
For more details you may try to increase the verbosity up to "-v3".
[ERR] JSON DS file: The anyxml "p" is expected to be represented as JSON name/value, but input data contains name/empty array.
[ERR] Callback "load" of plugin "JSON DS file" for module "czechlight-roadm-device" failed.
```
I think that empty arrays are valid for `anyxml`, and these should not be rejected. If the data were invalid, it would make sense to reject the writeout (or indeed constructing them in the first place), not just the subsequent reading.

Also, I was surprised that the operational DS files in SHM are JSON-formatted; I thought that the LYB format was used for these transient-only data for performance reasons. Does it mean that each push of operational data now incurs JSON parsing over and over again?

**Comment by michalvasko** [comment]
Alright, the empty array should now be supported.

> Also, I was surprised that the operational DS files in SHM are JSON-formatted

Yes, this is a recent change when the LYB datastore was changed to JSON datastore. The performance differences are negligible and there were problems constantly being discovered in LYB until I finally gave up. It is still used for internal communication but not as a datastore because it has problems with changing context.
